inc/admin/namespace.php

```diff
@@ -245,6 +240,47 @@ function handle_edit_submit( Client $consumer = null ) {
 	exit;
 }
 
+function enqueue_javascript() {
+	wp_localize_script('oauth2-edit-application', 'oauth2_ajax', [
+		'action' => AJAX_GENERATE_ACCESS_TOKEN,
+		'url' => admin_url( 'admin-ajax.php' ),
+	]);
+	wp_enqueue_script( 'oauth2-edit-application' );
+}
+
+function get_application_id() {
+	if ( ! empty( $_SERVER['HTTP_REFERER'] ) ) {
+		$id = \parse_url( $_SERVER['HTTP_REFERER'], PHP_URL_QUERY );
+		$params = [];
+		parse_str( $id, $params );
+		if ( ! isset( $params['id'] ) ) {
+			throw 'Not on admin page of the Application';
+		} else {
+			return $params['id'];
+		}
+	}
+
+	throw 'No server referer';
+}
+
+function generate_test_access_token() {
+	$client = Client::get_by_post_id( get_application_id() );
+	$token = $client->issue_token( wp_get_current_user() );
+
+	if ( is_wp_error( $token ) ) {
+		throw $token->getMessage();
+	}
+
+	$data = [
+		'access_token' => $token->get_key(),
+		'token_type'   => 'bearer',
+	];
+
+	echo \json_encode( $data );
```

Comment:
Unneeded `\`. Also, we should either use `echo wp_json_encode( $data )` or `wp_send_json( $data )` (which will `die` for us here.

Potentially, we could use a REST API endpoint for this instead?